src/java/org/apache/cassandra/config/DatabaseDescriptor.java

```diff
@@ -1830,11 +1857,57 @@ public static void setInterDCStreamThroughputOutboundMegabitsPerSec(int value)
         conf.inter_dc_stream_throughput_outbound_megabits_per_sec = value;
     }
 
-    public static String[] getAllDataFileLocations()
+    /**
+     * Checks if the local system data must be stored in a specific location which supports redundancy.
+     *
+     * @return {@code true} if the local system keyspaces data must be stored in a different location,
+     * {@code false} otherwise.
+     */
+    public static boolean useSpecificLocationForLocalSystemData()
+    {
+        return conf.local_system_data_file_directory != null;
+    }
+
+    /**
+     * Returns the locations where the local system keyspaces data should be stored.
+     *
+     * <p>If the {@code local_system_data_file_directory} was unspecified, the local system keyspaces data should be stored
+     * in the first data directory. This approach guarantees that the server can tolerate the lost of n - 1 disks.</p>
+     *
+     * @return the locations where should be stored the local system keyspaces data
+     */
+    public static String[] getLocalSystemKeyspacesDataFileLocations()
```

Comment:
We could return a single `String` instead of an array, since it is always a single directory. I gave it a go and doing so doesn't seem problematic with the current callers of the method, and `CassandraDaemon#migrateSystemDataIfNeeded()` is explicitly extracting the first element of the array.

Reply:
I hit some issues with that when writing the initial patch when the data directories where empty. Some tests run with such a config. I came up with that approach because it was allowing the code to behave in the same way as before the patch.
Unfortunately, I do not remeber all the details. I would have to try again if you think it is a better approach to use a String.

Reply:
I see the problem with tests, `DatabaseDescriptor#getLocalSystemKeyspacesDataFileLocations` is using an empty array to represent the fact that `data_file_directories` hasn't been set. [Here](https://github.com/adelapena/cassandra/commit/93bbe2e3a983811cce91c3ded989346ee8ea7c4d) is what I tried, returning a single `String` that can be null if `data_file_directories` is empty. Managing this situation inside the constructor of `DataDirectories` makes those tests to pass ([here](https://app.circleci.com/pipelines/github/adelapena/cassandra/167/workflows/aed26434-3657-47fa-a3a5-c5b20eced2aa)), while having a `DatabaseDescriptor#getLocalSystemKeyspacesDataFileLocation` method returning a nullable string (it could also be an optional). Nevertheless I think this is not a big deal and I guess we can just return an array in `getLocalSystemKeyspacesDataFileLocations`.